### PR TITLE
Update registerTransaction matcher usage - Closes #70

### LINKED ISSUE #70: Don’t use transaction type at the time of transaction registration
### Expected behavior

Transaction type should not be used as unique identifier for a transaction registration

In the long run we need to get rid of transaction type as identifying key. 

Related issue in lisk-sdk: https://github.com/LiskHQ/lisk-sdk/issues/3523

### Actual behavior

Current implementation of transaction registration uses `type` as identifier when registering transactions.

### Which version(s) does this affect? (Environment, OS, etc...)
2.0

### PR DESCRIPTION
**Pull request by mitsujutsu**
### What was the problem?

`registerTransaction` was modified in `lisk-sdk` framework. Changes need to be reflected in core.

### How did I fix it?

Updated `registerTransaction` calls in core.

### Review checklist

* The PR resolves #70 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

**Comment by shuse2** [comment]
@lsilvs Yes for now until the first release

**Comment by shuse2** [comment]
@lsilvs @mitsujutsu My bad, this PR was actually for 2.1, so it needs to be targeted to development.
I will create development branch now

File: src/transactions/5_dapp_transaction.ts [hunk 1]
```diff
@@ -318,7 +307,7 @@ export class DappTransaction extends BaseTransaction {
 		const errors: TransactionError[] = [];
 		const nameExists = store.transaction.find(
 			(transaction: TransactionJSON) =>
-				transaction.type === TRANSACTION_DAPP_TYPE &&
+				transaction.type === this.type &&
```
Comment (lsilvs) [inline src/transactions/5_dapp_transaction.ts]:
```suggestion
				transaction.type === this.TYPE &&
```

File: src/transactions/5_dapp_transaction.ts [hunk 2]
```diff
@@ -336,7 +325,7 @@ export class DappTransaction extends BaseTransaction {
 
 		const linkExists = store.transaction.find(
 			(transaction: TransactionJSON) =>
-				transaction.type === TRANSACTION_DAPP_TYPE &&
+				transaction.type === this.type &&
```
Comment (lsilvs) [inline src/transactions/5_dapp_transaction.ts]:
```suggestion
				transaction.type === this.TYPE &&
```

**Pull request merged**